### PR TITLE
Code Style - New line before and after method parentheses

### PR DESCRIPTION
**Pull request by ericdecanini**
Code style change to have new line before and after method parentheses
 
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

`ij_kotlin_method_parameters_new_line_after_left_paren`
and
`ij_kotlin_method_parameters_right_paren_on_new_line`

Both code style changes set to true

## Motivation and context

Closes https://github.com/vector-im/element-android/issues/5771

## Screenshots / GIFs

N/A

## Tests

- Smoke test the app works as normal

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 12

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,901 +1,901 @@
...
+ij_kotlin_keep_first_column_comment = true
+ij_kotlin_keep_indents_on_empty_lines = false
+ij_kotlin_keep_line_breaks = true
+ij_kotlin_lbrace_on_next_line = false
+ij_kotlin_line_comment_add_space = false
+ij_kotlin_line_comment_at_first_column = true
+ij_kotlin_method_annotation_wrap = split_into_lines
+ij_kotlin_method_call_chain_wrap = off
+ij_kotlin_method_parameters_new_line_after_left_paren = true
+ij_kotlin_method_parameters_right_paren_on_new_line = true
```
Comment (ericdecanini) [inline .editorconfig:814]: The actual changes

File: library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt [hunk 2]
```diff
@@ -44,7 +44,7 @@ class SwipeToDismissHandler(
     @SuppressLint("ClickableViewAccessibility")
     override fun onTouch(v: View, event: MotionEvent): Boolean {
         when (event.action) {
-            MotionEvent.ACTION_DOWN                          -> {
+            MotionEvent.ACTION_DOWN -> {
```
Comment (ericdecanini) [inline library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt:47]: It just so happens that Kotlin files with multiple classes at root level don't get affected by when arrow alignment settings (and always autoformats) this way.

As we're making this change too, should we just leave it like this? @bmarty

Comment (bmarty) [reply library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt:47]: I am a bit bored with this rule about arrow alignment. Can we start by changing this one? i.e. not align arrows anymore? Can be a quick and small PR I think.

Comment (ericdecanini) [reply library/attachment-viewer/src/main/java/im/vector/lib/attachmentviewer/SwipeToDismissHandler.kt:47]: Sounds good. Can do

File: .editorconfig (hunk 1 above)
Comment (bmarty) [reply .editorconfig:814]: OK. I think we should first ensure that the CI checks the formatting before enabling more rules. WDYT?

**Comment by ericdecanini** [comment]
Do we need a changelog file for this?

File: vector/src/main/java/im/vector/app/features/login2/LoginWebFragment2.kt [hunk 3]
```diff
@@ -124,8 +124,10 @@ class LoginWebFragment2 @Inject constructor(
         views.loginWebWebView.loadUrl(url)
 
         views.loginWebWebView.webViewClient = object : WebViewClient() {
-            override fun onReceivedSslError(view: WebView, handler: SslErrorHandler,
-                                            error: SslError) {
+            override fun onReceivedSslError(
+                    view: WebView, handler: SslErrorHandler,
```
Comment (Claire1817) [inline vector/src/main/java/im/vector/app/features/login2/LoginWebFragment2.kt:128]: One line per parameter ? :)

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt [hunk 4]
```diff
@@ -83,7 +83,8 @@ class ExportEncryptionTest {
     @Test
     fun checkExportDecrypt1() {
         val password = "password"
-        val input = "-----BEGIN MEGOLM SESSION DATA-----\nAXNhbHRzYWx0c2FsdHNhbHSIiIiIiIiIiIiIiIiIiIiIAAAACmIRUW2OjZ3L2l6j9h0lHlV3M2dx\n" + "cissyYBxjsfsAndErh065A8=\n-----END MEGOLM SESSION DATA-----"
+        val input =
+                "-----BEGIN MEGOLM SESSION DATA-----\nAXNhbHRzYWx0c2FsdHNhbHSIiIiIiIiIiIiIiIiIiIiIAAAACmIRUW2OjZ3L2l6j9h0lHlV3M2dx\n" + "cissyYBxjsfsAndErh065A8=\n-----END MEGOLM SESSION DATA-----"
```
Comment (yostyle) [inline matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt:87]: Concatenation is not necessary when strings are on the same line.
Same remark [here](https://github.com/vector-im/element-android/blob/3f3662e60562ca76f954c3d7f025a05c5524c584/matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt#L53)

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt [hunk 5]
```diff
@@ -83,7 +83,8 @@ class ExportEncryptionTest {
     @Test
     fun checkExportDecrypt1() {
         val password = "password"
-        val input = "-----BEGIN MEGOLM SESSION DATA-----\nAXNhbHRzYWx0c2FsdHNhbHSIiIiIiIiIiIiIiIiIiIiIAAAACmIRUW2OjZ3L2l6j9h0lHlV3M2dx\n" + "cissyYBxjsfsAndErh065A8=\n-----END MEGOLM SESSION DATA-----"
+        val input =
```
Comment (yostyle) [inline matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt:86]: Why did you add a line break here ?
You didn't add it [L53](https://github.com/vector-im/element-android/blob/3f3662e60562ca76f954c3d7f025a05c5524c584/matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt#L53) and [L68](https://github.com/vector-im/element-android/blob/3f3662e60562ca76f954c3d7f025a05c5524c584/matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt#L68)

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt [hunk 6]
```diff
@@ -103,7 +104,8 @@ class ExportEncryptionTest {
     @Test
     fun checkExportDecrypt2() {
         val password = "betterpassword"
-        val input = "-----BEGIN MEGOLM SESSION DATA-----\nAW1vcmVzYWx0bW9yZXNhbHT//////////wAAAAAAAAAAAAAD6KyBpe1Niv5M5NPm4ZATsJo5nghk\n" + "KYu63a0YQ5DRhUWEKk7CcMkrKnAUiZny\n-----END MEGOLM SESSION DATA-----"
+        val input =
+                "-----BEGIN MEGOLM SESSION DATA-----\nAW1vcmVzYWx0bW9yZXNhbHT//////////wAAAAAAAAAAAAAD6KyBpe1Niv5M5NPm4ZATsJo5nghk\n" + "KYu63a0YQ5DRhUWEKk7CcMkrKnAUiZny\n-----END MEGOLM SESSION DATA-----"
```
Comment (yostyle) [inline matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt:108]: Concatenation is not necessary when strings are on the same line.

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt [hunk 7]
```diff
@@ -123,7 +125,8 @@ class ExportEncryptionTest {
     @Test
     fun checkExportDecrypt3() {
         val password = "SWORDFISH"
-        val input = "-----BEGIN MEGOLM SESSION DATA-----\nAXllc3NhbHR5Z29vZG5lc3P//////////wAAAAAAAAAAAAAD6OIW+Je7gwvjd4kYrb+49gKCfExw\n" + "MgJBMD4mrhLkmgAngwR1pHjbWXaoGybtiAYr0moQ93GrBQsCzPbvl82rZhaXO3iH5uHo/RCEpOqp\nPgg29363BGR+/Ripq/VCLKGNbw==\n-----END MEGOLM SESSION DATA-----"
+        val input =
+                "-----BEGIN MEGOLM SESSION DATA-----\nAXllc3NhbHR5Z29vZG5lc3P//////////wAAAAAAAAAAAAAD6OIW+Je7gwvjd4kYrb+49gKCfExw\n" + "MgJBMD4mrhLkmgAngwR1pHjbWXaoGybtiAYr0moQ93GrBQsCzPbvl82rZhaXO3iH5uHo/RCEpOqp\nPgg29363BGR+/Ripq/VCLKGNbw==\n-----END MEGOLM SESSION DATA-----"
```
Comment (yostyle) [inline matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt:129]: Concatenation is not necessary when strings are on the same line.

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt [hunk 8]
```diff
@@ -123,7 +125,8 @@ class ExportEncryptionTest {
     @Test
     fun checkExportDecrypt3() {
         val password = "SWORDFISH"
-        val input = "-----BEGIN MEGOLM SESSION DATA-----\nAXllc3NhbHR5Z29vZG5lc3P//////////wAAAAAAAAAAAAAD6OIW+Je7gwvjd4kYrb+49gKCfExw\n" + "MgJBMD4mrhLkmgAngwR1pHjbWXaoGybtiAYr0moQ93GrBQsCzPbvl82rZhaXO3iH5uHo/RCEpOqp\nPgg29363BGR+/Ripq/VCLKGNbw==\n-----END MEGOLM SESSION DATA-----"
+        val input =
```
Comment (yostyle) [inline matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt:128]: Why did you add a line break here ?

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt [hunk 9]
```diff
@@ -202,7 +205,8 @@ class ExportEncryptionTest {
 
     @Test
     fun checkExportEncrypt4() {
-        val password = "passwordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpassword" + "passwordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpassword"
+        val password =
```
Comment (yostyle) [inline matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt:208]: Why did you add a line break here ?

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt [hunk 10]
```diff
@@ -202,7 +205,8 @@ class ExportEncryptionTest {
...
+                "passwordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpassword" + "passwordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpassword"
```
Comment (yostyle) [inline matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt:209]: Concatenation is not necessary when strings are on the same line.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/store/db/query/SharedSessionQueries.kt [hunk 11]
```diff
@@ -48,12 +50,14 @@ internal fun SharedSessionEntity.Companion.get(realm: Realm, roomId: String?, se
...
 }
 
-internal fun SharedSessionEntity.Companion.create(realm: Realm, roomId: String?,
-                                                  sessionId: String,
-                                                  userId: String,
-                                                  deviceId: String,
-                                                  deviceIdentityKey: String,
-                                                  chainIndex: Int): SharedSessionEntity {
+internal fun SharedSessionEntity.Companion.create(
+        realm: Realm, roomId: String?,
```
Comment (yostyle) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/store/db/query/SharedSessionQueries.kt:54]: you should add a line break between parameters.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/verification/VerificationTransportRoomMessage.kt [hunk 12]
```diff
@@ -163,8 +167,10 @@ internal class VerificationTransportRoomMessage(
     override fun cancelTransaction(transactionId: String, otherUserId: String, otherUserDeviceIds: List<String>, code: CancelCode) =
```
Comment (yostyle) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/verification/VerificationTransportRoomMessage.kt:167]: You should probably add line breaks between parameters.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/database/helper/ThreadEventsHelper.kt [hunk 13]
```diff
@@ -49,7 +49,8 @@ internal fun Map<String, EventEntity>.updateThreadSummaryIfNeeded(
         roomId: String,
         realm: Realm, currentUserId: String,
```
Comment (yostyle) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/database/helper/ThreadEventsHelper.kt:50]: You should add a line break between parameters.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/BindThreePidsTask.kt [hunk 14]
```diff
@@ -33,11 +33,13 @@ internal abstract class BindThreePidsTask : Task<BindThreePidsTask.Params, Unit>
...
 
-internal class DefaultBindThreePidsTask @Inject constructor(private val profileAPI: ProfileAPI,
-                                                            private val identityStore: IdentityStore,
-                                                            @AuthenticatedIdentity
-                                                            private val accessTokenProvider: AccessTokenProvider,
-                                                            private val globalErrorReceiver: GlobalErrorReceiver) : BindThreePidsTask() {
+internal class DefaultBindThreePidsTask @Inject constructor(
+        private val profileAPI: ProfileAPI,
+        private val identityStore: IdentityStore,
+        @AuthenticatedIdentity
```
[the remark this answers is not in the document]
Comment (ericdecanini) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/BindThreePidsTask.kt:39]: "A single annotation without arguments may be placed on the same line as the corresponding declaration:" is what the style guide says. It seems down to personal preference.

Imo I like them on the same line but there's no way to do that as far as the editorconfig goes to my knowledge and testing. Either way, this change and decision would have to be done separately from this pr

File: vector/src/main/java/im/vector/app/features/login2/LoginWebFragment2.kt (hunk 3 above)
Comment (ericdecanini) [reply vector/src/main/java/im/vector/app/features/login2/LoginWebFragment2.kt:128]: I've not found a way to do this either via editorconfig. If they were already previously formatted like this, they'd stay this way even after autoformatting.

I separated the method parameters you brought up in your comments but I can't do this app-wide

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt (hunk 5 above)
Comment (ericdecanini) [reply matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/ExportEncryptionTest.kt:86]: This is the auto-formatter, presumably because the line went over the char limit.

I manually changed this and the below remark to be multiline (the proper way), but be wary it may still be possible we see this elsewhere in the app

**Comment by ericdecanini** [comment]
To summarise the last 2 commits and the remarks that led up to them, let me summarise:

1) Method parameters that were previously on the same line will still be on the same line post-formatting. To my knowledge, we can't fix this via the editor config and we should just fix them manually as we see them.

2) Some strings that are very long and go over the 160 char limit may be autoformatted to be placed on a new line even if it concatenates or is still over the char limit after this line break. Again I manually fixed these instances, but they are remnants of going over the char limit in the first place and to my knowledge. also can't be fixed automatically via editor config

**Review by Claire1817 — APPROVED** [review]

**Review by ouchadam — APPROVED** [review]

File: .editorconfig (hunk 1 above)
Comment (ouchadam) [reply .editorconfig:814]: unfortunately we'll need to move away from the large set of fine grained editorconfig rules and switch to relying on only ktlint's rulesets, then we'll be able to run `./gradlew ktlintFormat` and check for uncommitted formatting 

I'd push for applying/merging the currently agreed upon updated rules and then look to enforce consistency with ktlint on the CI

(unless we want to use intellij as a 815mb standalone formatter :fearful:)

**Review by yostyle — APPROVED** [review]
LGTM
